### ISSUE TITLE
Consistency on the read side

### ISSUE THREAD

**Issue by johanandren** [akka/akka-persistence-couchbase#51]
In PR #44 we enabled REQUEST_PLUS as consistency for the read side to make the tests pass. This is not a requirement for read sides in general, since they are eventually consistent, so adding the overhead of this consistency just to make the test pass is a bit weird.

**Comment by patriknw** [comment]
I agree, queries can be eventually consistent. Replay of events on recovery must see the last event written as covered by issue #53

**Comment by johanandren** [comment]
I think the tricky part is that the query TCK expects the result to be consistent, haven't looked into if there are any TCK-knobs to tweak yet though.

**Comment by patriknw** [comment]
feel free to change the tck, such as using `awaitAssert` or `eventually`

**Comment by johanandren** [comment]
I think we need some read behind here, to make sure all written tagged events on all nodes has reached the index when we query so that we are sure we are delivering the events in the right order.

Do we also need a sequence number so that we can detect gaps. I'm not entirely sure I understand why those are needed in the C* journal @chbatey ?

**Comment by chbatey** [comment]
They are used to detect gaps for a persistenceId and them go back and look for them. 

For C* we can receive the events from the tags table in a different order than we wrote them. 

Now we have implemented eventual-consistency delay they are less useful as it gives time for the events to get there but it still allows quite strong guarantees even when running with a 0s delay as we detect gaps right away and go searching.
